File: backend/Rashtan.AuditStory.PaymentGizmo/ServiceCollectionExtension.cs
using Microsoft.Extensions.DependencyInjection;

namespace Rashtan.AuditStory.PaymentGizmo
{
    public static class ServiceCollectionExtension
    {
        public static void RegisterPaymentGizmo(this IServiceCollection services)
        {
            services.AddSingleton<IPaymentGizmo, PaymentGizmo>();
        }
    }
}

File: backend/Rashtan.AuditStory.Payment/ServiceCollectionExtension.fs
namespace Rashtan.AuditStory.Payment

open System.Runtime.CompilerServices
open Microsoft.Extensions.DependencyInjection

[<Extension>]
type ServiceCollectionExtension =
    [<Extension>]
    static member RegisterPaymentGizmo(services: IServiceCollection) =
        services.AddSingleton<IPaymentGizmo, PaymentGizmo>()
